### PR TITLE
Fix undefined self (FontUtils.js)

### PR DESCRIPTION
**Pull request by AtiX**
Fix error caused by undefined self when using three on the node.js command line (e.g. in tests)

### CODE REVIEW

File: src/extras/FontUtils.js [hunk 1]
```diff
@@ -464,5 +464,10 @@ THREE.FontUtils.generateShapes = function ( text, parameters ) {
 } )( THREE.FontUtils );
 
 // To use the typeface.js face files, hook up the API
-self._typeface_js = { faces: THREE.FontUtils.faces, loadFace: THREE.FontUtils.loadFace };
-THREE.typeface_js = self._typeface_js;
+var typeface_js = { faces: THREE.FontUtils.faces, loadFace: THREE.FontUtils.loadFace };
+if ( typeof self != 'undefined' ){
```
Comment (mrdoob) [inline src/extras/FontUtils.js]: I would do `if ( self !== undefined )` to follow the rest of the code base.

**Comment by mrdoob** [comment]
@zz85 Looks good?

File: src/extras/FontUtils.js (hunk 1 above)
Comment (AtiX) [inline src/extras/FontUtils.js]: Funny thing is, this also causes an error ('Reference Error: self is not defined') on my node installation - I would suggest to keep it this way

Comment (mrdoob) [inline src/extras/FontUtils.js]: Uh... node...

**Comment by zz85** [comment]
looks fine to me. could do `self !== 'undefined'` to enforce strict equality.

if there are more places to check whether script runs in browser / node, we could have a variable somewhere that does `THREE.NODE_ENV = typeof window === 'undefined'` too

**Comment by AtiX** [comment]
@zz85 replaced with `!==`

I am not sure about the node_env variable, since I don't think the there is a need for the code to know what environment it is running in. This pull request rather tries to make the code work on node without manual adjustments :)

File: src/extras/FontUtils.js [hunk 2]
```diff
@@ -464,5 +464,10 @@ THREE.FontUtils.generateShapes = function ( text, parameters ) {
 } )( THREE.FontUtils );
 
 // To use the typeface.js face files, hook up the API
-self._typeface_js = { faces: THREE.FontUtils.faces, loadFace: THREE.FontUtils.loadFace };
-THREE.typeface_js = self._typeface_js;
+var typeface_js = { faces: THREE.FontUtils.faces, loadFace: THREE.FontUtils.loadFace };
+if ( typeof self !== 'undefined' ){
+	self._typeface_js = typeface_js
```
Comment (mrdoob) [inline src/extras/FontUtils.js]: One last thing. Missing ; in this line.

File: src/extras/FontUtils.js [hunk 3]
```diff
@@ -464,5 +464,10 @@ THREE.FontUtils.generateShapes = function ( text, parameters ) {
...
+}
+THREE.typeface_js = typeface_js
```
Comment (mrdoob) [inline src/extras/FontUtils.js]: Missing ; also in this line.

**Comment by AtiX** [comment]
@mrdoob anything left to do?

**Comment by mrdoob** [comment]
All good! Thanks!

**Pull request merged**